src/fish_indent.cpp

```diff
@@ -251,9 +252,28 @@ void prettifier_t::prettify_node(const parse_node_tree_t &tree, node_offset_t no
                 if (prev_node_type != parse_token_type_redirection) {
                     append_whitespace(node_indent);
                 }
-                output.append(source, node.source_start, node.source_length);
+                wcstring unescaped{source, node.source_start, node.source_length};
+                // Unescape the string - this leaves special markers around if there are any expansions or anything.
+                // TODO: This also already computes backslash-escapes like \u or \x.
+                // We probably don't want that - if someone picked `\x20` to spell space, they have a reason.
+                unescape_string_in_place(&unescaped, UNESCAPE_SPECIAL);
+
+                // Remove INTERNAL_SEPARATOR because that's a quote.
+                auto quote = [](const wchar_t ch) {
```

Comment:
I'd not use `const` when passing by value (if that still compiles).

Reply:
Yeah, compiles.